### PR TITLE
Farm Section Update

### PR DESCRIPTION
**Pull request by sofinico**
## Description 

Impacts on https://docs.alexlab.co/. This branch is live at a temporary URL [here](https://coinfabrik.gitbook.io/alexlab-doc/v/alexlab-doc-farming/product-features/farming).

1. The "PRODUCT/FEATURES -> Farm" section was updated in content and in structure with a new Features template. For more details on the template, see Appendix. 
    - **Note:** FAQs answers will be completed on next PR (self-service farming).

2. The section was also renamed from "Farm" to "Farming" for a more descriptive title of the feature.

3. The `.gitbook.yaml` was updated to redirect old paths to renamed ones.

## Dev notes

Please merge #3 first. The changes relative to this PR are from https://github.com/alexgo-io/alexlab-doc/pull/4/commits/7d5c5e1d17a412eadb2066f1753666b44f4976f1 ahead. 

## Documentation Update Roadmap (so far)

- [x] Liquidity Providers 
- [x] Token Swaps 
- [x] Farming 
- [ ] Self Service Farming
- [ ] Stake
- [ ] Launchpad
- [ ] Analytics
- [ ] Self Service Listing
- [ ] New structure 

### Features template

This template is designed for the Features of ALEX Lab App. It is intended to go on https://docs.alexlab.co/ Gitbook.

```
FEATURES                    # Gitbook Group
├── Feature 1               # Gitbook Section with Cards
│ ├── How to                # Gitbook Subsection
│ ├── Key concepts          # Gitbook Subsection
│ └── FAQs                  # Gitbook Subsection with Expandables
├── Feature 2
│ ├── How to
│ ├── Key concepts
│ └── FAQs
└── ...
```

See what are Gitbook [Cards](https://docs.gitbook.com/content-editor/blocks/cards) and [Expandable sections](https://docs.gitbook.com/content-editor/blocks/expandable). The idea is that Sections are visible on the side-bar but subsections are only visible when you are reading the section.

Each page's content is described below.

#### Feature

Provides an overview of the feature. It should include a TL;DR paragraph summarizing the feature and providing links to the three nested sections (Key Concepts, How To, FAQs). This page should be useful for all types of users: newcomers may be directed to the how-to guides, while intermediate and advanced users may refer directly to specific FAQs or key concepts.

#### Key concepts

Brief explanation of key concepts related to the feature. We should try to be brief and concise here, as the user wants easy information. Diagrams and images are encouraged to enhance understanding.

#### How to

Contains:

- Introductory paragraph briefly summarizing the main idea behind the feature, with the intention to frame the reader and anticipate what will find in the guide.
- Step-by-step guide: A detailed how-to guide with screenshots, if possible, to assist users in following along.
- A practical example to illustrate the steps or concepts discussed (optional).

#### FAQs

The Frequently Asked Questions section addresses specific questions that users might have. Key concepts can also be explained in a Q&A format here, which allows the Key Concepts section to remain brief.

### CODE REVIEW

**Review by tenuki — APPROVED** [review]

File: product-features/farming/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    visible: true
+  outline:
+    visible: true
+  pagination:
+    visible: false
+---
+
+# 🌾 Farming
+
+Yield Farming on ALEX is an excellent way to earn exciting returns. By farming (staking or locking up your LP tokens), you earn rewards in addition to the earnings from being a liquidity provider. Among these rewards are $ALEX and APower tokens.
```
Comment (oscartsui) [inline product-features/farming/README.md:18]: Among these rewards are $ALEX,  APower tokens, and tokens provided by the project.

File: product-features/farming/how-to.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+</div>
+
+### Step 7
+
+After successfully staking your LP tokens in a farm, you will be able to see your active farms in the "My Farms" panel on the main [Farms page](https://app.alexlab.co/farm).
+
+<figure><img src="../../.gitbook/assets/farming/my-farms.png" alt=""><figcaption><p>Example of the "My Farms" panel. Here you will find all your active farms; click on any of them for detailed information.</p></figcaption></figure>
+
+By clicking on a farm, you will access the "My Farming" dashboard for that specific farm, which includes detailed metrics. On the right side of the dashborad, you will see that the current cycle has no earnings and no farming tokens. This is due to the cooldown period, farming will start in the next upcoming cycle. This is why it's convenient to stake for long periods: every time you stake, you must wait for the current cycle to end before you start generating rewards in the next cycle.
```
Comment (oscartsui) [inline product-features/farming/how-to.md:157]: Original sentence [This is due to the cooldown period, farming will start in the next upcoming cycle.]
Suggestion:
The reason you can't join the current reward cycle is that it had already started prior to your participation. However, once you successfully stake your LP into the farm, it gets registered for the next cycle. This assures you a proportional share of the farm rewards based on the number of LP tokens you have staked.

File: product-features/farming/how-to.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+For example, you can manually stake your \$ALEX rewards to generate compounding interest. If your rewards are another token, you can still [swap](https://app.alexlab.co/swap) and convert them to \$ALEX. Another option would be to use your rewards to buy more LP tokens and benefit from being a [liquidity provider](../liquidity-providers/README.md).
+
+Happy Farming! 🥕 🥬 🍅
+
+## :bellhop: Withdrawing LP tokens (unstake)
+
+Withdrawing you LP tokens takes just a few steps. The important thing is when to do it.
+
+When farming, you are commit to locking up your tokens for a predefined period (reward cycles, each one ~3.5 days). Once these cycles conclude, you will be able to unstake them and regain control over your LP tokens.
```
Comment (oscartsui) [inline product-features/farming/how-to.md:192]: When farming, you are committing to locking up your tokens for a predefined period (reward cycles, each cycle contains 525 stacks blocks, an estimation of ~3.5 days per cycle). Once these cycles conclude, you will be able to unstake them and regain control over your LP tokens.

File: product-features/farming/how-to.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+If you staked multiple times at different moments, the lock periods apply to each amount separately.
+
+Let's go through it step-by-step:
+
+1. Go to the [Farms page](https://app.alexlab.co/farm) on ALEX Lab App, which you can access through the Earn -> Farm tab.
+2. Find the farm from which you want to withdraw LP tokens and click on it.
+3. You will see the **My Farming** dashboard. Expand it to see all your farming details.
```
Comment (oscartsui) [inline product-features/farming/how-to.md:200]: 2. You will see the **My Farming** dashboard. Expand it to see all your farming details.
3. Find the farm from which you want to withdraw LP tokens and click on it.

File: product-features/farming/how-to.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
+---
+description: >-
+  Step-by-step guides to learn how to stake LP tokens into a farm, harvest
+  rewards and remove LP tokens from the farm.
+---
+
+# How to farm & harvest
+
+Yield farming takes a few easy steps to get set up. It works very similar to standard staking, with the key difference being that the tokens you stake are LP tokens. As in traditional staking, you lock up your tokens for a certain period (measured in cycles) and earn rewards over time. After each (except for the first cooldown one), you will have rewards available to harvest.
```
Comment (Hadan-ALEX) [inline product-features/farming/how-to.md:9]: "As in traditional staking" -> "As with traditional staking"

File: product-features/farming/how-to.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+5. Click on the "Harvest All" button and confirm the transaction in your wallet (just as you did in the previous guides). This will return your LP tokens back to your possession and automatically collect any unharvested rewards.
+6. Wait for the transaction to be confirmed on the network. Remember, you can turn on [Telegram notifications](https://t.me/stacks\_tx\_notification\_bot) or search for the transaction on the [ALEX Explorer](https://app.alexlab.co/explorer).
+7. Once the transaction is completed, you will see the changes reflected in your wallet balance and on the platform panels. In particular, you will see your LP token balance on the [Pool page](https://app.alexlab.co/pool) in the **My Liquidity** panel or by selecting the pool from the list. You can also check your balance on the ALEX Lab App, located beside the Wallet Manager at the top menu bar.
+
+If you want to farm your LP tokens again, remember: your staked amount will start generating yield from the next upcoming cycle, as the current cycle is in cooldown period. To maximize the APR you earn, it's best to stake for longer cycle periods to avoid missing out on any reward cycles due to the cooldown period.
```
Comment (oscartsui) [inline product-features/farming/how-to.md:206]: If you want to farm your LP tokens again, remember: your staked amount will start generating yield from the next upcoming cycle.

File: product-features/farming/how-to.md (hunk 5 above)
Comment (Hadan-ALEX) [inline product-features/farming/how-to.md:9]: "As in traditional staking," -> "As with traditional staking,"

"After each (except..." -> "After each cycle (except..."

File: product-features/farming/how-to.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+### Step 3
+
+Once you have selected the farm, enter the amount of LP tokens you would like to stake, or click "Max" to use all available LP Tokens.&#x20;
+
+Next, choose the number of reward cycles you want to lock your tokens into the farm. Each cycle is approximately 3.5 days. 
+
+<figure><img src="../../.gitbook/assets/farming/add-lp-staking.png" alt="" width="375"><figcaption><p>Example with an amount of LP tokens to stake for 32 cycles.</p></figcaption></figure>
+
+{% hint style="info" %}
+Your staked amount will start generating yield from the next upcoming cycle, as the current cycle is in "cooldown" period. To maximize the APR you earn, it's best to stake for longer cycle periods to avoid missing out on any reward cycles due to this cooldown cycle. That's why 32-cycle staking is recommended.
```
Comment (Hadan-ALEX) [inline product-features/farming/how-to.md:91]: "That's why 32-cycle staking is recommended."

"That's why 32-cycle staking is recommended if your goal is to maximize earnings."

File: product-features/farming/key-concepts.md [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+* **Trading Pair:** The specific LP token that the farm accepts. To obtain these LP tokens, you will have to provide liquidity to the pool associated with the same trading pair.
+* **Token Rewards:** The token in which the farm rewards the stakers at the end of each cycle. This token is predefined by the farm.
+
+Farms only accept LP tokens of one kind. For example, the STX-ALEX farm only accepts STX-ALEX LP tokens, which you receive in exchange for providing liquidity to the STX-ALEX pool.
+
+### Cycles and cooldown period
+
+Farming is measured in cycles. **One cycle** is 525 Stacks blocks, which is approximately **3.5 days**. This means that when you stake tokens in a farm, you need to specify the number of cycles you want to lock up your tokens in the farm. Rewards are distributed after a cycle ends.
+
+Your staked tokens will start generating yield on the next upcoming cycle, meaning that there is a **cooldown period** between the time you stake and the start of a new cycle. To maximize your earnings, it's best to stake for longer cycle periods, avoiding gaps in rewards due to the cooldown. That's why 32-cycle staking is recommended.
```
Comment (oscartsui) [inline product-features/farming/key-concepts.md:36]: Your staked tokens will start generating yield in the next upcoming cycle, which means there will be no reward during the time gap between when you stake and when the upcoming cycle starts.

File: product-features/farming/key-concepts.md [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+
+Let's put it on an example. Say you stake for 1 cycle at a time. When the cycle ends, you can claim the rewards associated to that cycle. To keep generating rewards, you will have to withdraw your LP tokens and restake them. But when you stake them again, that current cycle is in cooldown, so you will have to wait for the next one to obtain rewards. Over 100 cycles, this method would cause you to miss rewards for about 50 cycles. In contrast, if you stake for 32 cycles, you will only miss rewards for 3 cycles.
```
Comment (oscartsui) [inline product-features/farming/key-concepts.md:38]: Let's use an example. Assume you stake for one cycle at a time. When that cycle ends, you can claim the rewards associated with that cycle. To continue generating rewards, you will have to withdraw your LP tokens and restake them. However, when you restake them, the current cycle will not be eligible for you to earn rewards. Therefore, you will have to wait until the next cycle to acquire rewards.

File: product-features/farming/key-concepts.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+### Reward distribution
+
+Farms may offer different types of reward tokens, and each farm has a predetermined amount of rewards. For simplicity, we can assume that the total rewards distributed to stakers during each cycle remains constant. At the end of each cycle, the rewards are available to be harvested by the farmers (stakers).
```
Comment (oscartsui) [inline product-features/farming/key-concepts.md:42]: At the end of each cycle, the rewards are available to be harvested by the farmers (stakers) proportionate to the LP tokens shares within that cycle.

File: product-features/farming/key-concepts.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+Rewards are distributed proportionally to each farmer based on their staked amount. This can be represented by the following equation:
+
+$$
+\begin{equation} \textrm{Farmer Reward} = \frac{\textrm{Farmer Staked Amount}}{\textrm{Total Staked Amount}} \; \cdot \; \textrm{Total Rewards}. \end{equation}
+$$
+
+Each value in the equation applies to a specific cycle.
+
+When there are two reward tokens (e.g. $ALEX and APower), the formula is applied separately for each token, resulting in two Farmer Reward amounts, one for each reward token.
```
Comment (oscartsui) [inline product-features/farming/key-concepts.md:52]: When there are two reward tokens (e.g. $ALEX and APower), the formula is applied separately for each token, resulting in two Farming Reward amounts, one for each reward token.

File: product-features/liquidity-providers/how-to.md [hunk 12]
```diff
@@ -43,14 +43,14 @@ After selecting a pool, you will be taken to a control panel for that specific l
...
 
 {% hint style="warning" %}
 Clicking the "Max" button will automatically set the amount to your total available balance.
 {% endhint %}
 
-[^1]: The APR metric is the same as the displayed in the Fee Rebate column on the previous step.
-
 ### Step 4
 
 If you want to adjust slippage, select the "Settings" icon to open the Transaction Settings panel and set your desired tolerance. The default slippage tolerance is set to 4%, meaning your transaction will revert if the exchange rate changes unfavourably by more than this percentage. The displayed number of LP tokens you will receive is approximate due to this potential variation.
```
Comment (oscartsui) [inline product-features/liquidity-providers/how-to.md:56]: If you want to adjust slippage, select the "Settings" icon to open the Transaction Settings panel and set your desired tolerance. The default slippage tolerance for non-stable swap token pairs is set to 4%, meaning your transaction will revert if the exchange rate changes unfavourably by more than this percentage. The displayed number of LP tokens you will receive is approximate due to this potential variation.

Comment (oscartsui) [reply product-features/liquidity-providers/how-to.md:56]: I am planning to change the default slippage setting of stable swap pair from 4% to 0.5%

File: product-features/farming/README.md (hunk 1 above)
Comment (arturoBeccar) [reply product-features/farming/README.md]: Corrected

File: product-features/farming/how-to.md (hunk 2 above)
Comment (arturoBeccar) [reply product-features/farming/how-to.md]: Corrected

File: product-features/farming/how-to.md (hunk 3 above)
Comment (arturoBeccar) [reply product-features/farming/how-to.md]: Corrected

File: product-features/farming/how-to.md (hunk 5 above)
Comment (arturoBeccar) [reply product-features/farming/how-to.md]: Corrected

File: product-features/farming/how-to.md (hunk 6 above)
Comment (arturoBeccar) [reply product-features/farming/how-to.md]: Corrected

File: product-features/farming/how-to.md (hunk 5 above)
Comment (arturoBeccar) [reply product-features/farming/how-to.md]: Corrected

File: product-features/farming/how-to.md (hunk 7 above)
Comment (arturoBeccar) [reply product-features/farming/how-to.md]: Corrected

File: product-features/farming/key-concepts.md (hunk 8 above)
Comment (arturoBeccar) [reply product-features/farming/key-concepts.md]: Corrected

File: product-features/farming/key-concepts.md (hunk 9 above)
Comment (arturoBeccar) [reply product-features/farming/key-concepts.md]: Corrected

File: product-features/farming/key-concepts.md (hunk 10 above)
Comment (arturoBeccar) [reply product-features/farming/key-concepts.md]: Corrected and rephrased:
At the end of each cycle, the rewards are available to be harvested by the farmers (stakers) in proportion to their share of LP tokens within that cycle.

File: product-features/farming/key-concepts.md (hunk 11 above)
Comment (arturoBeccar) [reply product-features/farming/key-concepts.md]: Corrected, added a missing comma and added bold letters

**Pull request merged**